store/tikv/gc_worker.go

```diff
@@ -43,10 +45,111 @@ type GCWorker struct {
 	lastFinish  time.Time
 	cancel      goctx.CancelFunc
 	done        chan error
+
+	session tidb.Session
+}
+
+// SafePointKV is used for a seamingless integration for mockTest and runtime.
+type SafePointKV interface {
+	Put(k string, v string) error
+	Get(k string) (string, error)
+}
+
+// MockSafePointKV implements SafePointKV at mock test
+type MockSafePointKV struct {
+	store    map[string]string
+	mockLock sync.RWMutex
+}
+
+// NewMockSafePointKV creates an instance of MockSafePointKV
+func NewMockSafePointKV() *MockSafePointKV {
+	return &MockSafePointKV{
+		store: make(map[string]string),
+	}
+}
+
+// Put implements the Put method for SafePointKV
+func (w *MockSafePointKV) Put(k string, v string) error {
+	w.mockLock.Lock()
+	defer w.mockLock.Unlock()
+	w.store[k] = v
+	return nil
+}
+
+// Get implements the Get method for SafePointKV
+func (w *MockSafePointKV) Get(k string) (string, error) {
+	w.mockLock.RLock()
+	defer w.mockLock.RUnlock()
+	elem, ok := w.store[k]
+	if !ok {
+		return "", nil
+	}
+	return elem, nil
+}
+
+// EtcdSafePointKV implements SafePointKV at runtime
+type EtcdSafePointKV struct {
+	cli *clientv3.Client
+}
+
+// NewEtcdSafePointKV creates an instance of EtcdSafePointKV
+func NewEtcdSafePointKV(addrs []string) (*EtcdSafePointKV, error) {
+	etcdCli, err := createEtcdKV(addrs)
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+	return &EtcdSafePointKV{cli: etcdCli}, nil
+}
+
+// Put implements the Put method for SafePointKV
+func (w *EtcdSafePointKV) Put(k string, v string) error {
+	ctx, cancel := goctx.WithTimeout(goctx.Background(), time.Second*5)
+	_, err := w.cli.Put(ctx, k, v)
+	cancel()
+	if err != nil {
```

Comment:
You can return `errors.Trace(err)` directly here

Reply:
Thanks~